pacsman/filesystem_dev_client.py

```diff
@@ -144,10 +147,19 @@ def date_filter(study_ds):
             else:
                 study_date_str = None
 
-            if study_start_date is None or study_end_date is None or study_date_str is None:
+            if (study_start_date is None and study_end_date is None) or study_date_str is None:
```

Comment:
This is a small thing, but can we disambiguate the names `study_start_date`, `study_end_date`, and `study_date_str` a bit more visually? They all look pretty similar and it makes it a bit difficult to quickly parse this logic. I think it would be helpful to convey with variable names that `study_start_date` and `study_end_date` are part of the query, and `study_date_str` is the attribute of the study being filtered.

I have liked to use the word "candidate" to distinguish this in the past, although I bet there are better ways to do this too. e.g. `candidate_study_date` instead of `study_date_str`.

Reply:
Renamed in `e04c8a175337593392e63cbbeb32390448f76628`. I used `raw_*` for the unparsed date strings, which I think is somewhat common for untrusted user input or to indicate an unparsed input.